SMBLibrary/Client/SMB2Client.cs

```diff
@@ -415,7 +428,7 @@ private void ProcessPacket(SessionPacket packet, ConnectionState state)
                 {
                     SMB2TransformHeader transformHeader = new SMB2TransformHeader(packet.Trailer, 0);
                     byte[] encryptedMessage = ByteReader.ReadBytes(packet.Trailer, SMB2TransformHeader.Length, (int)transformHeader.OriginalMessageSize);
-                    messageBytes = SMB2Cryptography.DecryptMessage(m_decryptionKey, transformHeader, encryptedMessage);
+                    messageBytes = SMB2Cryptography.DecryptMessage(m_decryptionProvider, transformHeader, encryptedMessage);

```

Comment:
I understand that we want to use different provider, what I don't understand is changing the DecryptMessage method signature

Reply:
Sure, there's a bit more going on in the DecryptMessage method (extracting the headers, associated data, etc..) that I was trying not to duplicate. 

I wanted to use the same instance of the AesCcm class for a given key as creating/tearing them down involves a bit of overhead, and to do that needed to pass in that additional state. Since this was an internal class I thought It might be more appropriate to change this signature to and remove the selection of the actual cryptographic implementation from this method. 

I can do this differently if you think it is more appropriate.